### ISSUE TITLE
403 Error when updating thumbnails (AH01626: authorization result of Require all denied)

### ISSUE THREAD

**Issue by kcaran** [shaarli/Shaarli#1549]
Hi,

I was having an issue trying to synchronize thumbnails on my server. It would try to update the thumbnail for the first link and I would get a 403 error alert. I tried following the Apache 2.4 configuration documentation and turned on debugging:

```
[Wed Sep 09 14:00:44.100697 2020] [ssl:debug] [pid 9244] ssl_engine_kernel.c(225
): [client 12.130.32.5:39154] AH02034: Subsequent (No.3) HTTPS request received
for child 2 (server shaarli.kcaran.com:443), referer: https://shaarli.kcaran.com
/admin/tools
[Wed Sep 09 14:00:44.101126 2020] [authz_core:debug] [pid 9244] mod_authz_core.c
(809): [client 12.130.32.5:39154] AH01626: authorization result of Require all g
ranted: granted, referer: https://shaarli.kcaran.com/admin/tools
[Wed Sep 09 14:00:44.101137 2020] [authz_core:debug] [pid 9244] mod_authz_core.c
(809): [client 12.130.32.5:39154] AH01626: authorization result of <RequireAny>:
 granted, referer: https://shaarli.kcaran.com/admin/tools
[Wed Sep 09 14:00:44.101266 2020] [authz_core:debug] [pid 9244] mod_authz_core.c
(809): [client 12.130.32.5:39154] AH01626: authorization result of Require all g
ranted: granted, referer: https://shaarli.kcaran.com/admin/tools
[Wed Sep 09 14:00:44.101275 2020] [authz_core:debug] [pid 9244] mod_authz_core.c
(809): [client 12.130.32.5:39154] AH01626: authorization result of <RequireAny>:
 granted, referer: https://shaarli.kcaran.com/admin/tools
[Wed Sep 09 14:00:44.370217 2020] [ssl:debug] [pid 9244] ssl_engine_kernel.c(225
): [client 12.130.32.5:39154] AH02034: Subsequent (No.4) HTTPS request received
for child 2 (server shaarli.kcaran.com:443), referer: https://shaarli.kcaran.com
/admin/thumbnails
[Wed Sep 09 14:00:44.370622 2020] [authz_core:debug] [pid 9244] mod_authz_core.c
(728): [client 12.130.32.5:39154] AH01625: authorization result of Require all g
ranted: neutral (directive limited to other methods), referer: https://shaarli.k
caran.com/admin/thumbnails
[Wed Sep 09 14:00:44.370649 2020] [authz_core:debug] [pid 9244] mod_authz_core.c
(809): [client 12.130.32.5:39154] AH01626: authorization result of Require all d
enied: denied, referer: https://shaarli.kcaran.com/admin/thumbnails
[Wed Sep 09 14:00:44.370654 2020] [authz_core:debug] [pid 9244] mod_authz_core.c
(809): [client 12.130.32.5:39154] AH01626: authorization result of <RequireAny>:
 denied, referer: https://shaarli.kcaran.com/admin/thumbnails
[Wed Sep 09 14:00:44.370658 2020] [authz_core:error] [pid 9244] [client 12.130.3
2.5:39154] AH01630: client denied by server configuration: /var/www/shaarli/admi
n, referer: https://shaarli.kcaran.com/admin/thumbnails
```

The issue is with the PATCH request to update the thumbnail:

```
2020-09-09 14:00:44 PATCH /admin/shaare/3237/update-thumbnail 443 12.130.32.5 HT
TP/1.1 "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/537.36 (KHTM
L, like Gecko) Chrome/85.0.4183.83 Safari/537.36" https://shaarli.kcaran.com/adm
in/thumbnails 403 669 555 575
```

I was able to fix it by adding PATCH to the root .htaccess file:

```
diff --git a/.htaccess b/.htaccess
index 4c00427..6683200 100644
--- a/.htaccess
+++ b/.htaccess
@@ -15,7 +15,7 @@ RewriteCond %{REQUEST_FILENAME} !-f
 RewriteCond %{REQUEST_FILENAME} !-d
 RewriteRule ^ index.php [QSA,L]

-<Limit GET POST PUT DELETE OPTIONS>
+<Limit GET POST PATCH PUT DELETE OPTIONS>
   <IfModule version_module>
     <IfVersion >= 2.4>
        Require all granted
@@ -31,7 +31,7 @@ RewriteRule ^ index.php [QSA,L]
   </IfModule>
 </Limit>

-<LimitExcept GET POST PUT DELETE OPTIONS>
+<LimitExcept GET POST PATCH PUT DELETE OPTIONS>
   <IfModule version_module>
     <IfVersion >= 2.4>
        Require all denied
```

Looking at the .htaccess file, I'm wondering how mine is the only instance with this issue. I'm not even sure what the .htaccess is trying to accomplish in having both a <Limit> and a <LimitExcept> clause.

**Comment by ArthurHoaro** [comment]
Thanks for the report!

> Looking at the .htaccess file, I'm wondering how mine is the only instance with this issue.

The PATCH route has been added with `v0.12.0-beta` which only a few people are running yet, but also `.htaccess` file only affects instances hosted behind Apache web server, while our Docker image uses nginx. That's probably why you were the first to find out.

> I'm not even sure what the .htaccess is trying to accomplish in having both a and a clause.

I had to dig a little and found the original addition here: https://github.com/shaarli/Shaarli/pull/1141#issuecomment-394524826
I think the idea is to allow specified HTTP methods, and deny access for any other.
